src/ChannelServer/World/Map.cs

```diff
@@ -251,6 +251,112 @@ public List<ICombatEntity> GetAttackableEntitiesInRange(ICombatEntity attacker,
 			return result;
 		}
 
+		/// <summary>
+		/// Returns attackable monsters sorted by their distance in the given 
+		/// radius around position.
+		/// </summary>
+		/// <param name="attacker"></param>
+		/// <param name="position"></param>
+		/// <param name="radius"></param>
+		/// <returns></returns>
+		public List<ICombatEntity> GetAttackableEntitiesInRangeSortedByDistance(Character attacker, Position position, float radius)
```

Comment:
The result wouldn't be sorted correctly if an entity is able to attack both monsters _and_ characters, would it? Because you're first adding the monsters sorted by range and then the characters.

Reply:
These methods aren't really supposed to be a part of the buff PR, they are a part of the skills branch, but the concern is correct regarding behavior, there isn't a superset of entities, so I presume I either add them to the list and then sort it.

Reply:
Removed skill-related changes, they aren't supposed to be a part of this PR.